pkg/statistics/histogram.go

```diff
@@ -1049,22 +1073,26 @@ func (hg *Histogram) OutOfRangeRowCount(
 		upperBound = hg.NotNullCount() / float64(histNDV)
 	}
 
-	allowUseModifyCount := sctx.GetSessionVars().GetOptObjective() != variable.OptObjectiveDeterminate
-
 	if !allowUseModifyCount {
 		// In OptObjectiveDeterminate mode, we can't rely on the modify count anymore.
 		// An upper bound is necessary to make the estimation make sense for predicates with bound on only one end, like a > 1.
 		// We use 1/NDV here (only the Histogram part is considered) and it seems reasonable and good enough for now.
 		return min(rowCount, upperBound)
 	}
 
-	// If the modifyCount is large (compared to original table rows), then any out of range estimate is unreliable.
+	// If the realtimeRowCount is larger than the original table rows, then any out of range estimate is unreliable.
```

Comment:
`If the realtimeRowCount is larger than the original table rows, then any out of range estimate is unreliable.` Is this assumption too strict? As long as the data size grow, this assumption will be satisfied.

Reply:
@qw4990 - you are correct that any changes which result in an increase in realtimeRowCount will cause a change from this code. However - for this code to be entered - you must be searching for values outside of the range of the current histograms. If the predicates do not search outside this range - then there is no impact from this change.

Using an example such as 20% would miss those situations where a small number of rows were added - similar to the example that I shared above.